pymc_marketing/clv/distributions.py

```diff
@@ -908,12 +908,12 @@ def rng_fn(cls, rng, alpha, beta, size):
         alpha = np.broadcast_to(alpha, size)
         beta = np.broadcast_to(beta, size)
 
-        p = rng.beta(a=alpha, b=beta, size=size)
+        p_samples = rng.beta(a=alpha, b=beta, size=size)
 
-        samples = rng.geometric(p, size=size)
-
-        return samples
+        # prevent log(0) by clipping small p samples
```

Comment:
I wouldn't do this. If the only reason is the test, relax it or seed it. I can easily make a test that still fails with this hack.